### ISSUE TITLE
support for other plot sources

### ISSUE THREAD

**Issue by daslu** [dvdt/gyptis#6]
Continuing the discussion in #5, it would be nice to support other plot sources like 
[gorilla-plot](http://gorilla-repl.org/plotting.html) and [vega-lite](https://vega.github.io/vega-lite/).

At least for gorilla-plot (which generates vega plots, too), it turns out to be quite easy (needs some adaptation of the resulting vega structure, maybe due to inconsistencies across vega versions).

I would be happy to prepare a pull request if you wish.

Edit: fixed issue reference.
Edit: formatting.

**Comment by dvdt** [comment]
This sounds great! A pull request would be most welcome.

I can think of a few separate avenues for making gyptis easy to use with vega-lite. Was wondering if you had any of these in mind or a different way altogether.
1. Run vega-lite in the browser. This would entail including vega-lite.js in the client. In javascript we would call `vl.compile` to compile a vega-lite spec into a vega spec. The existing workflow for rendering a vega spec in the browser could then be used. We could also create a new function called, say, `plot-vega-lite!` to use this facility.
2. Run vega-lite.js within the JVM, via Nashorn or Rhino? vega-lite specs would then be compiled within the JVM into vega. The existing client code would not need to be modified at all. I haven't been able to find anything about whether Nashorn can run vega-lite, so it may not be possible to do this. 
3. Create a new repo that provides clojure/script bindings to vega-lite. Let's say for example this repo is called `clj-vega-lite`, and it has only 1 function: `clj-vega-lite/compile` that transforms a vega-lite spec into a vega spec. In the clj context, `clj-vega-lite/compile` is done by Nashorn; and in cljs, it calls out to vega-lite.js. If such a library existed, then vega-lite could easily be used within gyptis. It might look something like:

``` clojure
(-> my-vega-lite-spec
    clj-vega-lite/compile
    gyptis.view/plot!)
```

Thoguhts?

**Comment by daslu** [comment]
Thanks!
1. seems the easiest (and has the advantage of passing less data through the socket).
2. should be interesting to try, one day (I do not have any experience with Nashorn or Rhino).
3. is interesting for its extensibility - it has the potential of becoming something more powerful, that still has vega-lite as a subset.

If it is ok with you, I will try Option 1 for now.

By the way, it can be simplified a little by using [vega-embed](https://github.com/vega/vega-embed), which [supports vega-lite format](https://vega.github.io/vega-lite/usage/embed.html) - it is only necessary to specify the `mode`.

Edit: minor fix for accuracy

**Comment by dvdt** [comment]
Agreed that option 1 seems to be the easiest. Looking forward to seeing what you come up with!
